### DISCUSSION TITLE
[@types/amplitude-js] is missing transport

### DISCUSSION THREAD

**Discussion by jonalu** [DefinitelyTyped/DefinitelyTyped#57574] [Issues with a @types package]
## The problem:
According to the [Amplitude Developer Portal](https://developers.amplitude.com/docs/advanced-settings#using-sendbeacon) the SDK version 8.5.0 supports sending events using the browser's built-in navigator.sendBeacon API.

The type definition supports neither of these approaches:
```
// set transport to 'beacon' when initializing an event
amplitude.getInstance().init('API_KEY', 'USER_ID', {transport: 'beacon'});

// set transport to 'beacon' after initialization
amplitude.getInstance().setTransport('beacon');
```

The following should be added:
- Interface `amplitude.Config` should define property `transport: Transport`
- Class `AmplitudeClient` should define method `setTransport(transport: Transport)`
- The [different supported transports](https://github.com/amplitude/Amplitude-JavaScript/blob/acf596a52ff12bad8c9352533fd15f13becf374a/src/constants.js#L65-L66) _should_ be a separate type `Transport`

[a comment by an automated account is not shown]

**Comment by danmana** [comment]
Added a PR
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57576
